src/containers/audio-rewards-page/components/SendInputBody.tsx

```diff
@@ -54,49 +61,92 @@ type BalanceError =
   | 'INSUFFICIENT_TRANSFER_AMOUNT'
   | 'EMPTY'
   | 'MALFORMED'
-type AddressError = 'MALFORMED' | 'EMPTY' | 'SEND_TO_SELF'
+  | 'LESS_THAN_MIN'
+
+type AddressError =
+  | 'MALFORMED'
+  | 'EMPTY'
+  | 'SEND_TO_SELF'
+  | 'INVALID_SPL_ADDRESS'
+
+const makeMinAudioError = (num: number | string) =>
+  `You must send at least ${num} $AUDIO`
 
 const balanceErrorMap: { [B in BalanceError]: string } = {
   INSUFFICIENT_BALANCE: messages.insufficientBalance,
   EMPTY: messages.amountRequired,
   MALFORMED: messages.amountMalformed,
-  INSUFFICIENT_TRANSFER_AMOUNT: messages.amountInsufficient
+  INSUFFICIENT_TRANSFER_AMOUNT: messages.amountInsufficient,
+  LESS_THAN_MIN: 'LESS_THAN_MIN' // special case this key to create on the fly
 }
 
 const addressErrorMap: { [A in AddressError]: string } = {
   MALFORMED: messages.addressMalformed,
   EMPTY: messages.addressRequired,
-  SEND_TO_SELF: messages.addressIsSelf
+  SEND_TO_SELF: messages.addressIsSelf,
+  INVALID_SPL_ADDRESS: messages.validSPLAddress
 }
 
 type SendInputBodyProps = {
   currentBalance: BNWei
-  onSend: (balance: BNWei, destinationAddress: WalletAddress) => void
+  onSend: (
+    balance: BNWei,
+    destinationAddress: WalletAddress,
+    chain: Chain
+  ) => void
   wallet: WalletAddress
+  solWallet: WalletAddress
 }
 
-const isValidDestination = (wallet: WalletAddress) => {
+const isValidEthDestination = (wallet: WalletAddress) => {
   const libs = window.audiusLibs
   return libs.web3Manager.web3.utils.isAddress(wallet)
 }
 
+const isValidSolDestination = (wallet: WalletAddress) => {
+  const solanaweb3 = window.audiusLibs.solanaWeb3Manager.solanaWeb3
+  try {
+    const _ = new solanaweb3.PublicKey(wallet)
```

Comment:
why the var?

Reply:
This will just error if it's not a valid public key